### PR TITLE
Disable feature when it would cause an error with expression trees.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/16028

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationDiagnosticAnalyzer.cs [hunk 1]
```diff
@@ -217,6 +226,28 @@ private void AnalyzeSyntaxNode(SyntaxNodeAnalysisContext context)
                 additionalLocations: allLocations));
         }
 
+        private bool IsInExpressionTree(
+            SemanticModel semanticModel, SyntaxNode argumentExpression, CancellationToken cancellationToken)
+        {
+            var expressionType = semanticModel.Compilation.GetTypeByMetadataName("System.Linq.Expressions.Expression`1");
```
Comment (Pilchie) [inline src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationDiagnosticAnalyzer.cs]: Is this worth caching in a `CompilationStarted` callback?

Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationDiagnosticAnalyzer.cs]: Probably is.  I'll make that change.  Thanks!

**Comment by CyrusNajmabadi** [comment]
Tagging @srivatsn .  This fixes an issue with one of our CodeFixers.  The reason this is important to fix is:

1. People want to get to a 'clean' state wrt to analyzer diagnostics.  So having analyzers tell them they should fix things that would actually break code is bad.
2. This negatively affects people who say 'fix all'.  The fix here will break code and force users to undo the change.

**Pull request merged**